sympy/polys/polytools.py

```diff
@@ -1678,6 +1681,7 @@ def pexquo(f, g):
         ExactQuotientFailed: 2*x - 4 does not divide x**2 + 1
 
         """
+        g = sympify(g)
         _, per, F, G = f._unify(g)
```

Comment:
Aren't these cases handled by `_unify_`?

Reply:
They were, but sometimes that leads to double sympification. And it seems like it makes sense that "private" functions do not sympify.

Reply:
I think that `_unify_` is different even if it is an internal call because it's purpose is essentially the same as sympify. Potentially some of these shouldn't use sympify though. They expect Poly rather than general Expr anyway e.g.:
```python
In [7]: Poly(x**2).pexquo(x)
---------------------------------------------------------------------------
CoercionFailed

In [16]: Poly(x**2).pexquo(Poly(x))
Out[16]: Poly(x, x, domain='ZZ')
```
I'm not sure exactly what the set of acceptable inputs here is if it is more than just `int/Poly`. It seems like it will maybe  accept any Expr that coerces to the ground domain e.g.:
```python
In [13]: Poly(x**2*y, x).pexquo(y)
Out[13]: Poly(y**3*x**2, x, domain='ZZ[y]')
```
There are separate top-level functions that take arbitrary Expr and then call these methods internally:
```python
In [14]: pexquo(x**2, x)
Out[14]: x
```
It's not really clear what the expected behaviour is.

Reply:
OK! I just went with the idea of sympify in fewer locations and `_unify` was really the outlier when it came to consequences. However, there will be fewer sympify calls in total now, although on more lines.

But it is of course possible to revert the change in that file if that is preferred.

Reply:
What causes the unnecessary sympify calls? How are you measuring this?

Reply:
In current master, add calls sympify first and then in `_unity`. 
https://github.com/sympy/sympy/blob/0bc2587aef09ab150bd408b7cd9e08baf7b5fd9c/sympy/polys/polytools.py#L1446-L1451

After this change, all calls are moved from `_unity` to the method calling `_unity`, so no redundant calls.

(Then I noted the _polifyit decorator here, so maybe there are still redundant sympifications, but beyond my knowledge if it should be possible to use e.g. the add method separately or if the overloading of + is enough.)

Reply:
I don't completely understand the design of `__add__` vs `add`. I would have made it so that only `__add__` sympifies but then when you know you have two Polys you can call `add`.

Reply:
Agreed. So the question is really if someone uses add instead? Maybe one should deprecated `add` and introduce `_add` instead? (Which is basically only called from `__add__`, or from library code where one know that it is sympified.)

Not in this PR though... (There are quite a lot of "public" functions and methods that clearly only are internal helpers. As you most likely know, but I was struck by it searching the code...)

Reply:
Maybe there should be a `_unify_sympify` function or conversely a `_unify_no_sympify` function. The main idea of the current design of `_unify` is to reduce repetition in these methods and the change here works against that.

Reply:
I reverted this file.

Reply:
But, yes, there is a discrepancy between redundant code and redundant calls.